web-server/src/components/PRTable/PullRequestsTableHead.tsx

```diff
@@ -85,216 +127,21 @@ export const PullRequestsTableHead: FC<PullRequestsTableHeadProps> = ({
             </DarkTooltip>
           </TableCell>
         )}
-        <TableCell sx={{ minWidth: '40%', p: CELL_PAD, py: 1.5 }}>
-          <TableSortLabel
-            direction={conf.field === 'number' ? conf.order : 'asc'}
-            active={conf.field === 'number'}
-            onClick={() => updateSortConf('number')}
-          >
-            Pull Request
-          </TableSortLabel>
-        </TableCell>
-
-        {enabledColumnsSet.has('commits') && (
-          <TableCell sx={{ minWidth: '40%', p: CELL_PAD, py: 1.5 }}>
+        {/* Map through the columnHeaders object to dynamically generate table headers */}
+        {Object.entries(columnHeaders).map(([key, label]) => (
+          <TableCell key={key} sx={{ minWidth: '40%', p: CELL_PAD, py: 1.5 }}>
             <TableSortLabel
-              direction={conf.field === 'commits' ? conf.order : 'asc'}
-              active={conf.field === 'commits'}
-              onClick={() => updateSortConf('commits')}
+              direction={conf.field === key ? conf.order : 'asc'}
+              active={conf.field === key}
+              onClick={() => updateSortConf(key as keyof PR)}
             >
-              Commits
+              {label}
             </TableSortLabel>
           </TableCell>
-        )}
+        ))}
```

Comment:
While this is concise, this makes the order of the column headers susceptible to unexpected change.

The order of object also appears to not be guaranteed to the order of insertion:
https://stackoverflow.com/a/5525820

You may use a `Map`.
Regardless, it still doesn't account for the fact that not every header had the exact same styling properties. Some had a minWidth, some had no styling, some had `nowrap`, some had centered alignment.

This combined with a lack of screenshots, and a lack of tests, makes me think you didn't actually properly test this PR at all.

In think for your purposes, the simplest overall change would be to use the property names (like `created_at`) as headers instead of using formatted names (like `Created At`). That way, you retain the type safety without touching any of the existing code.